Update "End-to-End Tissue Microarray Image Analysis with Galaxy-ME" tools, workflow, and tutorial

CameronFRWatson opened the pull request:
**changes**
- Updated tools, answer key history, and workflow
- Updated tutorial text, images, references

**To-do**
- Still need to create workflow tests; however, I wanted to get the PR up so that I might address any reviews and create the workflow tests in parallel

bgruening commented:
@CameronFRWatson the container does not work for us on EU. It seems you are using the `vitessce` user inside the container by default. But if you use a different user that has not the ID `1000` it will not work. How are you running this container? The last container was build differently? Can we revert to the previous owner by default?

CameronFRWatson commented:
Hi @bgruening. We hopefully have addressed this here: https://github.com/goeckslab/tools-mti/pull/95

We just updated the existing container so there shouldn't be any need to pull any changes to test. Let us know if you continue to run into problems

CameronFRWatson commented:
Question as I work on resolving some of the CI errors: 

Does the test data for the workflow need to be the same data used in the tutorial? I'm concerned that while the data used in the tutorial is good for learning these tools with a realistic use-case, it might be too resource and time consumptive for CI. Is it acceptable to use workflow test data that is either a subset of, or entirely different from the tutorial dataset?

shiltemann commented:
@CameronFRWatson no, you can use different test data if that makes more sense :+1:

shiltemann commented on topics/imaging/tutorials/multiplex-tissue-imaging-TMA/workflows/GTN_Exemplar_002_TMA_workflow_Feb2025.ga:
For this rename, you can now add an entry in the redirects folder so that the old link gets redirected (and to fix the build error)

https://github.com/galaxyproject/training-material/tree/main/redirects

CameronFRWatson replied:
thank you!

kostrykin commented on topics/imaging/tutorials/multiplex-tissue-imaging-TMA/tutorial.md:
```diff
@@ -167,35 +165,36 @@ UNetCoreograph will output images (used for downstream steps), masks, and a prev
-  - ilastik ({% cite Berg2019 %})
 
-In this tutorial, we use **Mesmer** because it tends to perform generally well on a diverse range of image types, and has a limited number of parameters to understand.
+In this tutorial, we use **Mesmer** because it tends to perform generally well on a diverse range of image types, and can be parameterized to suite different modalities. 
```
I'm confused by the term "parameterized" here, not sure what you mean. AFAIK parameterization is the procedure of introducing new parameters (*not* parameter values). Maybe you mean
```suggestion
In this tutorial, we use **Mesmer** because it tends to perform generally well on a diverse range of image types, and can be adapted to suite different modalities. 
```
or
```suggestion
In this tutorial, we use **Mesmer** because it tends to perform generally well on a diverse range of image types, and even a single set of parameters can suite different modalities. 
```
where one might also replace `set of parameters` by `set of parameter values` to be more precise?

kostrykin reviewed (commented):
Two general comments:

1. I noticed that several tools that you are mentioning in the tutorial seem to be not available in Galaxy-ME. I only checked the cancer.usegalaxy.org instance though:
    - Quantification
    - Analyze and visualize spatial multi-omics data with Squidpy
    - Create spatial scatterplot with Squidpy

    Or maybe I'm missing something here? Can something be done about it?

2. Markdown syntax seems to be not available inside `<warning-title>`:

    <img width="447" alt="Bildschirmfoto 2025-03-13 um 13 37 34" src="https://github.com/user-attachments/assets/280b3bc2-1c63-4767-87a5-f0975533d1cc" />

    For example, see line 79, but this occurs several times more. Maybe replace line 79 by just `<warning-title>Imaging platform differences</warning-title>` and the other occasions accordingly?

beatrizserrano commented on topics/imaging/tutorials/multiplex-tissue-imaging-TMA/tutorial.md:
```diff
@@ -167,35 +165,36 @@ UNetCoreograph will output images (used for downstream steps), masks, and a prev
 >    >
->    > The **np.squeeze** parameter is very important to select as `Yes` to make the output compatible with next steps
+>    > Make sure the **np.squeeze** parameter is set to `Yes` to make the output compatible with next steps.
+>    > This is the default behavoir. 
```
```suggestion
>    > This is the default behavior. 
```

beatrizserrano commented on topics/imaging/tutorials/multiplex-tissue-imaging-TMA/tutorial.md:
```diff
@@ -320,27 +321,32 @@ Some tools can cause the channel names in an OME-TIFF image to be lost. To fix t
 ## Initial visualization with **Avivator**
 
-For any `OME-TIFF` image in a Galaxy-ME history, there will be an option to view the image using **Avivator**. This is a great way to perform an initial inspection of an image for QC purposes before continuing with downstream steps. The **Avivator** window can be launched by expanding the dataset information in the history panel and clicking the link (Figure 7.).
+For any `OME-TIFF` image in a Galaxy-ME history, there will be an option to view the image using **Avivator**. This is a great way to perform an initial inspection of an image for QC purposes before continuing with downstream steps. The **Avivator** window can be launched by clicking the {% icon dataset-visualize %} **Dataset visualization** icon under an expanded dataset, which will open a page  ...
```
```suggestion
For any `OME-TIFF` image in a Galaxy-ME history, there will be an option to view the image using **Avivator**. This is a great way to perform an initial inspection of an image for QC purposes before continuing with downstream steps. The **Avivator** window can be launched by clicking the {% icon dataset-visualize %} **Dataset visualization** icon under an expanded dataset, which will open a page with visualization options. From here, click the **Display at Avivator link** (Figure 6).
```

beatrizserrano commented on topics/imaging/tutorials/multiplex-tissue-imaging-TMA/tutorial.md:
```diff
@@ -320,27 +321,32 @@ Some tools can cause the channel names in an OME-TIFF image to be lost. To fix t
-> <hands-on-title> View Images with Avivator </hands-on-title>
-> 1. Expand the dataset `ASHLAR`: OME-TIFF image to be viewed
-> 2. Click on *"display at Avivator"*
+Clicking the link will open a new window with the Avivator interactive dashboard. Here, users can select channels to view, adjust the channel contrast and colors for viewing, and zoom in to inspect the image (Figure 7.). 
```
```suggestion
Clicking the link will open a new window with the Avivator interactive dashboard. Here, users can select channels to view, adjust the channel contrast and colors for viewing, and zoom in to inspect the image (Figure 7). 
```

beatrizserrano commented on topics/imaging/tutorials/multiplex-tissue-imaging-TMA/tutorial.md:
```diff
@@ -389,12 +404,25 @@ Galaxy-ME includes additional tools from **Scimap** and tools from the **Squidpy
 >    > <comment-title>Ripley's L plot</comment-title>
 >    >
->    > **Squidpy** was used to calculate Ripley's L curves for each phenotype in core 2 of exemplar 2 (Figure 11.). This shows the overall organization of each phenotype in the tissue. If the curve for a given phenotype lies above the light grey null line (Example: Epithelial cells in Figure 11.), the phenotype is statistically significantly clustered. If the curve lies on the null line (Example: ...
+>    > **Squidpy** was used to calculate Ripley's L curves for each phenotype in core 2 of exemplar 2 (Figure 12.). This shows the overall organization of each phenotype in the tissue. If the curve for a given phenotype lies above the light grey null line (Example: Epithelial cells in Figure 11.), the phenotype is statistically significantly clustered. If the curve lies on the null line (Example: ...
```
```suggestion
>    > **Squidpy** was used to calculate Ripley's L curves for each phenotype in core 2 of exemplar 2 (Figure 12). This shows the overall organization of each phenotype in the tissue. If the curve for a given phenotype lies above the light grey null line (Example: Epithelial cells in Figure 11.), the phenotype is statistically significantly clustered. If the curve lies on the null line (Example: Myeloid lineage in Figure 11.), it's spatial distribution within the tissue is random. If the curve is underneath the null line (Example: T cells in Figure 11.), it's spatial distribution is statistically significantly dispersed.
```

beatrizserrano commented on topics/imaging/tutorials/multiplex-tissue-imaging-TMA/tutorial.md:
```diff
@@ -352,32 +358,41 @@ For any `OME-TIFF` image in a Galaxy-ME history, there will be an option to view
 >    > <comment-title>Neighborhood enrichment plot</comment-title>
 >    >
->    > **Squidpy** was used to calculate neighborhood enrichments for each phenotype in core 2 of exemplar 2 (Figure 10.). This shows which phenotypes co-locate most frequently within the tissue.
+>    > **Squidpy** was used to calculate neighborhood enrichments for each phenotype in core 2 of exemplar 2 (Figure 11.). This shows which phenotypes co-locate most frequently within the tissue.
```
```suggestion
>    > **Squidpy** was used to calculate neighborhood enrichments for each phenotype in core 2 of exemplar 2 (Figure 11). This shows which phenotypes co-locate most frequently within the tissue.
```

beatrizserrano commented on topics/imaging/tutorials/multiplex-tissue-imaging-TMA/tutorial.md:
```diff
@@ -389,12 +404,25 @@ Galaxy-ME includes additional tools from **Scimap** and tools from the **Squidpy
+>
+>    > <comment-title>Spatial scatterplot</comment-title>
+>    >
+>    > **Squidpy** was used to generate a spatial scatterplot of core 2 of exemplar 2 (Figure 13.). This is useful for visualizing the distribution of cell types, or any other categorical or continuous variable in the anndata file. 
```
```suggestion
>    > **Squidpy** was used to generate a spatial scatterplot of core 2 of exemplar 2 (Figure 13). This is useful for visualizing the distribution of cell types, or any other categorical or continuous variable in the anndata file. 
```

CameronFRWatson commented:
> Two general comments:
> 
> 1. I noticed that several tools that you are mentioning in the tutorial seem to be not available in Galaxy-ME. I only checked the cancer.usegalaxy.org instance though:
>    
>    * Quantification
>    * Analyze and visualize spatial multi-omics data with Squidpy
>    * Create spatial scatterplot with Squidpy
>    
>    Or maybe I'm missing something here? Can something be done about it?
> 2. Markdown syntax seems to be not available inside `<warning-title>`:
>     <img alt="Bildschirmfoto 2025-03-13 um 13 37 34" width="447" src="https://private-user-images.githubusercontent.com/6557139/422342586-280b3bc2-1c63-4767-87a5-f0975533d1cc.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDE4OTI5NjUsIm5iZiI6MTc0MTg5MjY2NSwicGF0aCI6Ii82NTU3MTM5LzQyMjM0MjU4Ni0yODBiM2JjMi0xYzYzLTQ3NjctODdhNS1mMDk3NTUzM2QxY2MucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDMxMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAzMTNUMTkwNDI1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZGI5ODUyODZiYzI2Zjc1YzUwMWU3MTc5ZmJhNjYzMGFiZmJhYTZmN2QxZjdiNzc3OGNiOGQ5MWMyMWExOWUyNSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.J4AE-ek5iuCKpAYwTFAHT5BcOuAFmHzzIIQgyKaxmcI">
>    For example, see line 79, but this occurs several times more. Maybe replace line 79 by just `<warning-title>Imaging platform differences</warning-title>` and the other occasions accordingly?

@kostrykin, I updated the name of the quantification tool to MCQUANT in the tutorial. This is on the shed and on the cancer instance. The two Squidpy tools are on the toolshed, they just got removed from the cancer instance tool panel accidentally. I'll get them re-installed asap. Also fixed the markdown issue
